g2p_programs/models/programs.py

```diff
@@ -112,6 +112,31 @@ def _default_journal_id(self):
     locked = fields.Boolean(default=False)
     locked_reason = fields.Char()
 
+    def toggle_active(self):
+        """
+        Overrides the default :meth:`toggle_active` to cancel
+        all `draft` and `to_approve` associated cycles and
+        'draft' and 'pending_validation' entitlements.
+
+        :return: toggle_active function of parent class
+        """
+        for rec in self:
+            # Cancel cycles and entitlements only if the program is active (for archiving)
+            if rec.active:
+                _logger.info("Archive Program: cancel cycles and entitlements.")
+                if rec.cycle_ids:
+                    entitlement_manager = rec.get_manager(self.MANAGER_ENTITLEMENT)
+                    # Get only `draft` and `to_approve` cycles
+                    cycles = rec.cycle_ids.filtered(
```

Comment:
Shouldn't we deactivate all entitlements even though the cycle is active ?

Reply:
Do you mean deactivate entitlements with approved cycles?

Reply:
Your comment below answered by question. I will work on this now.

Reply:
Fixed